api/client/trust.go

```diff
@@ -0,0 +1,323 @@
+package client
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/docker/distribution/digest"
+	"github.com/docker/docker/pkg/homedir"
+	flag "github.com/docker/docker/pkg/mflag"
+	"github.com/docker/docker/pkg/term"
+	"github.com/docker/docker/pkg/tlsconfig"
+	"github.com/docker/docker/registry"
+	"github.com/docker/notary/client"
+)
+
+var untrusted bool
+
+func addTrustedFlags(fs *flag.FlagSet) {
+	var trusted bool
+	if e := os.Getenv("DOCKER_NOTARY"); e != "" {
+		var err error
+		trusted, err = strconv.ParseBool(e)
+		if err != nil {
+			// treat any other value as true
+			trusted = true
+		}
+	}
+	fs.BoolVar(&untrusted, []string{"-untrusted"}, !trusted, "Use or update trusted repositories on push and pull")
+}
+
+func isTrusted(name string) bool {
+	// TODO(dmcgowan): whitelist specific names to be untrusted
+	// TODO(dmcgowan): canonicalize name before checking whitelist
+	return !untrusted
```

Comment:
why cant we just name the bool trusted

On Mon, Jul 13, 2015 at 4:33 PM, Derek McGowan <notifications@github.com>
wrote:

> In api/client/trust.go
> <https://github.com/docker/docker/pull/14546#discussion_r34523973>:
>
> > +	var trusted bool
> > +	if e := os.Getenv("DOCKER_NOTARY"); e != "" {
> > +		var err error
> > +		trusted, err = strconv.ParseBool(e)
> > +		if err != nil {
> > +			// treat any other value as true
> > +			trusted = true
> > +		}
> > +	}
> > +	fs.BoolVar(&untrusted, []string{"-untrusted"}, !trusted, "Use or update trusted repositories on push and pull")
> > +}
> > +
> > +func isTrusted(name string) bool {
> > +	// TODO(dmcgowan): whitelist specific names to be untrusted
> > +	// TODO(dmcgowan): canonicalize name before checking whitelist
> > +	return !untrusted
>
> This was flipped after initial implementation from trusted to untrusted.
> The function could probably be named better.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/14546/files#r34523973>.
>